hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/auditing_architecture.md

```diff
@@ -141,8 +152,19 @@ thread.
 
 ### class `NoopAuditor`
 
-This auditor creates spans which perform no auditing.
-It is very efficient and reliable.
+This auditor creates spans which doesn't do anything with the events.
+
+```xml
+<property>
+  <name>fs.s3a.audit.service.classname</name>
+  <value>org.apache.hadoop.fs.s3a.audit.impl.NoopAuditor</value>
+</property>
+```
+
+This is *not* the same as disabling auditing, as it still uses the `ActiveAuditManagerS3A` class
```

Comment:
Getting confused here a bit. I see you have removed the threadlocal field from NoopAuditManagerS3A. So ideally now it shouldn't cause the memory leak. But  again you are saying it still uses the ActiveAuditManagerS3A which is a source of memory leaks.

And at the same point, when auditing is disabled by default, we are creating stubAuditManager which in turn creates NoopAuditManagerS3A in new code.

Reply:
I think I got it. Just changing the classname will still use ActiveAuditManagerS3A but stubAuditManager will create NoopAuditManagerS3A which doesn't have threadlocal.

Reply:
there is a NoopAuditor, this is the one you could already enable, which generated `NoopAuditSpans`

but there was also a stub audit manager, which is the class s3afs created to handle all the audit integration for that instance. it was used in two places
* mock s3a FS instances in unit tests
* in s3afs initialize so even when the aws sdk was probing s3, the request processing would find something.

the change now then is to make that no op manager the default everywhere, with tests to verify this. the parameterisation where will cause a bit of backport conflict with branches which still have s3guard on, i will address that